### PR TITLE
TheMagicDrip new components

### LINKED ISSUE #14196: [Components] the_magic_drip
# the_magic_drip
## URLs
- https://docs.themagicdrip.com/api-reference/introduction
#### Prompt
List available campaigns. No props needed besides the app.
#### URLs
- https://docs.themagicdrip.com/api-reference/endpoint/get-v1campaign
#### Prompt
Add a single lead to a campaign. Include optional props for the campaign's parameters, such as company and LinkedIn URL.
#### URLs
- https://docs.themagicdrip.com/api-reference/endpoint/post-v1campaignleads
#### Prompt
Mark a given campaign as active or inactive. List available campaigns with async options, and use a boolean prop for the desired state.
#### URLs
- https://docs.themagicdrip.com/api-reference/endpoint/post-v1campaign-inactive
- https://docs.themagicdrip.com/api-reference/endpoint/post-v1campaign-inactive
#### Prompt
List available templates. No props needed besides the app.
#### URLs
- https://docs.themagicdrip.com/api-reference/endpoint/get-v1templates
#### Prompt
Emit new event when a campaign is created. Poll the "list campaigns" endpoint and emit an event only for campaigns with the `createdAt` field greater than the last date/time the source was run.
#### URLs
- https://docs.themagicdrip.com/api-reference/endpoint/get-v1campaign
#### Prompt
Emit new event when a template is created. Poll the "list templates" endpoint and emit an event only for templates with the `createdAt` field greater than the last date/time the source was run.
#### URLs
- https://docs.themagicdrip.com/api-reference/endpoint/get-v1templates

### PR DESCRIPTION
**Pull request by GTFalcao**
Closes #14196 

## Summary by CodeRabbit

- **New Features**
	- Introduced new properties for campaign management: `campaignId`, `linkedinUrl`, and `desiredState`.
	- Added methods for campaign operations, including listing campaigns, adding leads, and toggling campaign activation.
  
- **Bug Fixes**
	- Minor adjustments across several components to ensure consistent formatting with newlines added at the end of files. 

- **Documentation**
	- Updated summaries to reflect changes in functionality and structure where applicable.

### CODE REVIEW

File: components/the_magic_drip/actions/mark_campaign_active_or_inactive/mark_campaign_active_or_inactive.mjs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import app from "../../the_magic_drip.app.mjs";
+
+export default {
+  key: "the_magic_drip-mark-campaign-active-inactive",
+  name: "Mark Campaign Active or Inactive",
+  description: "Marks a campaign as active or inactive. [See the documentation](https://docs.themagicdrip.com/api-reference/endpoint/post-v1campaign-active)",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/the_magic_drip/actions/mark_campaign_active_or_inactive/mark_campaign_active_or_inactive.mjs:7]:
```suggestion
  version: "0.0.1",
```

File: components/the_magic_drip/actions/add_lead_to_campaign/add_lead_to_campaign.mjs [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import app from "../../the_magic_drip.app.mjs";
+
+export default {
+  key: "the_magic_drip-add-lead-to-campaign",
+  name: "Add Lead to Campaign",
+  description: "Add a lead to a campaign. [See the documentation](https://docs.themagicdrip.com/api-reference/endpoint/post-v1campaignleads)",
+  version: "0.0.{{ts}}",
```
Comment (luancazarine) [inline components/the_magic_drip/actions/add_lead_to_campaign/add_lead_to_campaign.mjs:7]:
```suggestion
  version: "0.0.1",
```

**Review by luancazarine — CHANGES_REQUESTED** [review]
Hi @GTFalcao, I just added some suggestions.

File: components/the_magic_drip/sources/common.mjs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+      items?.filter?.((item) => !savedIds.includes(this.getItemId(item))).forEach((item, index) => {
+        if ((!maxEmits) || (index < maxEmits)) {
+          this.$emit(item, {
+            id: this.getItemId(item),
+            ...this.getItemMetadata(item),
+          });
+        }
+        savedIds.push(this.getItemId(item));
+      });
```
[the remark this answers is not in the document]
Comment (GTFalcao) [reply components/the_magic_drip/sources/common.mjs:43]: No, this is intentional, in order not to emit too many retroactive events when first deploying the source, and avoid emitting the leftover events at a later date (instead emitting only new events as they happen).

**Review by luancazarine — DISMISSED** [review]
Hi @GTFalcao, LGTM! Ready for QA!

**Review by luancazarine — APPROVED** [review]

**Pull request merged**